src/executor/iExecutor.py

```diff
@@ -3,9 +3,16 @@
 
 
 class iExecutor(ABC):
-    def __init__(self, next: 'iExecutor'):
+
+    def __init__(self, next: 'iExecutor' = None):
```

Comment:
Was the plan to have the executors themselves call the next executor, rather than having some kind of administrator manage how the data was being transferred?
I think doing the former would force the pipeline to run linearly. Don't remember what we decided on for that